marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/entities/search/impl/CreatedOnFacetHandler.java

```diff
@@ -21,33 +21,113 @@
-        // Adding 1 day to end date to get docs harmonized on the end date as well.
-        LocalDate endDate = LocalDate.parse(data.getRangeValues().getUpperBound(), DATE_FORMAT)
-                .plusDays(1);
-        String endDateTime = endDate.atStartOfDay(ZoneId.systemDefault()).format(DATE_TIME_FORMAT);
+            case "Custom":
+                getCustomTimeWindow(data, facetInputs);
+                break;
+        }
+        return facetInputs.dateRange;
+    }
+
+    private void getTodayTimeWindow(CreatedOnFacetInputs facetInputs) {
+        facetInputs.startDateTime = facetInputs.startDate.toLocalDate().atStartOfDay(facetInputs.zoneId).format(DATE_TIME_FORMAT);
+        facetInputs.dateRange.put("startDateTime", facetInputs.startDateTime);
+        facetInputs.endDateTime = facetInputs.endDate.plusDays(1).toLocalDate().atStartOfDay(facetInputs.zoneId).format(DATE_TIME_FORMAT);
+        facetInputs.dateRange.put("endDateTime", facetInputs.endDateTime);
+    }
+
+    private void getThisWeekTimeWindow(CreatedOnFacetInputs facetInputs) {
+        facetInputs.startDateTime = facetInputs.startDate.plusDays((-1) * (facetInputs.startDate.getDayOfWeek().getValue() % 7)).format(DATE_TIME_FORMAT);
+        facetInputs.dateRange.put("startDateTime", facetInputs.startDateTime);
+        facetInputs.endDateTime = facetInputs.endDate.plusDays(1).toLocalDate().atStartOfDay(facetInputs.zoneId).format(DATE_TIME_FORMAT);
+        facetInputs.dateRange.put("endDateTime", facetInputs.endDateTime);
+    }
+
+    private void getThisMonthTimeWindow(CreatedOnFacetInputs facetInputs) {
+        facetInputs.startDate = facetInputs.startDate.with(TemporalAdjusters.firstDayOfMonth());
+        facetInputs.startDateTime = facetInputs.startDate.toLocalDate().atStartOfDay(facetInputs.zoneId).format(DATE_TIME_FORMAT);
+        facetInputs.dateRange.put("startDateTime", facetInputs.startDateTime);
+        facetInputs.endDateTime = facetInputs.endDate.plusDays(1).toLocalDate().atStartOfDay(facetInputs.zoneId).format(DATE_TIME_FORMAT);
+        facetInputs.dateRange.put("endDateTime", facetInputs.endDateTime);
+    }
 
-        return queryBuilder
-                .and(queryBuilder.rangeConstraint(Constants.CREATED_ON_CONSTRAINT_NAME, StructuredQueryBuilder.Operator.GE, startDateTime),
-                        queryBuilder.rangeConstraint(Constants.CREATED_ON_CONSTRAINT_NAME, StructuredQueryBuilder.Operator.LT, endDateTime));
+    private void getCustomTimeWindow(DocSearchQueryInfo.FacetData data, CreatedOnFacetInputs facetInputs) {
+        if(!data.getRangeValues().getLowerBound().isEmpty() && !data.getRangeValues().getUpperBound().isEmpty()) {
+            try {
+                facetInputs.startDateTime = ZonedDateTime.parse(data.getRangeValues().getLowerBound()).format(DATE_TIME_FORMAT);
+                facetInputs.endDateTime = ZonedDateTime.parse(data.getRangeValues().getUpperBound()).plusDays(1).format(DATE_TIME_FORMAT);
```

Comment:
It seems that FE now sends both date and time. Do we still need to add a day any more if that's the case?
Can we use startTime >= && <=endTime for comparison making this more accurate?

Since the FE still uses a DatePicker and not DateTimePicker maybe we can add the day over there. This would make the BE not deal with the wonky logic of adding days. Also, this would mean when we introduce a DateTimePicker in the FE we don't have to change the BE to support it.

Reply:
Date and time for entity properties. We still have a date picker for hub properties
It is more accurate doing < than <= as we don't know appropriate end time for a date.

The feature supports client zone offset. To work with zone it has to be date-time, So converted to date-time even if date picker is used

Reply:
I meant can we add the day in the FE since the only reason we need to add a day here is that we use a date picker and not a date-time picker. Since this seems to be a FE dependency maybe the hack should also be in the FE.

Moving the logic to the FE would make BE work correctly for all date-time ranges and if/when we add the date-time picker in the FE we dont have to make any changes since we already support the right thing.

Reply:
The date picker value will change, as we display the same object. Its not just FE, even if the endpoint is hit, it should return the same results no different from UI